src/models/node.ts

```diff
@@ -363,6 +430,112 @@ export class Node<N extends INodeBase, E extends IEdgeBase> implements INode<N,
     });
   }
 
+  setData(data: N): void;
+  setData(callback: (node: INode<N, E>) => N): void;
+  setData(arg: N | ((node: INode<N, E>) => N)) {
+    if (isFunction(arg)) {
+      this._data = (arg as (node: INode<N, E>) => N)(this);
+    } else {
+      this._data = arg as N;
+    }
+    this.notifyListeners();
+  }
+
+  patchData(data: Partial<N>): void;
+  patchData(callback: (node: INode<N, E>) => Partial<N>): void;
+  patchData(arg: Partial<N> | ((node: INode<N, E>) => Partial<N>)) {
+    let data: Partial<N>;
+
+    if (isFunction(arg)) {
+      data = (arg as (node: INode<N, E>) => Partial<N>)(this);
+    } else {
+      data = arg as Partial<N>;
+    }
+
+    patchProperties(this._data, data);
+
+    this.notifyListeners();
+  }
+
+  setPosition(position: INodeCoordinates | INodeMapCoordinates | INodePosition, isInner?: boolean): void;
+  setPosition(
+    callback: (node: INode<N, E>) => INodeCoordinates | INodeMapCoordinates | INodePosition,
+    isInner?: boolean,
+  ): void;
+  setPosition(
+    arg:
+      | INodeCoordinates
+      | INodeMapCoordinates
+      | INodePosition
+      | ((node: INode<N, E>) => INodeCoordinates | INodeMapCoordinates | INodePosition),
+    isInner?: boolean,
```

Comment:
I have to use the second argument here because sometimes we don't need to notify listeners on set position (e.g., on `SIMULATION_PROGRESS` in `orb-view.ts` because position handling is done through the simulation). However, this approach does not seem to be very clean. Do you have any ideas on how I could make it better?

Reply:
If we decide to leave it here, perhaps a better naming option would be `isInnerCall`? I will also update the interface accordingly when we define it :)

Reply:
Aha. Angular uses `dispatch: false` for those kind of events. `isInnerCall` doesn't really say what you described so I would go into the direction such as: `isNotifySkipped` or `isDispatchSkipped`, `isNotifyIgnored`. In general, use the name that describes what you described to me, you don't want to notify listeners. And the default should be false or not existing object, that's why I used `skipped` or `Ignored`. Feel free to come up with a name with such constraints.

Reply:
Also, this should be an object, optional `options?`.